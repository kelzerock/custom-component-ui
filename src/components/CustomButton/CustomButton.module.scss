:root {
  --color-main-blue: #1976d2;
  --color-main-blue-100: #2070c0;
  --color-white: #fff;
  --color-white-100: #f6fafd;
}

.button {
  margin: 8px;
  text-transform: uppercase;
  padding: 3px 9px;
  font-size: 0.8125rem;
  background-color: var(--color-white);
  color: var(--color-main-blue);
  font-weight: 500;
  outline: none;
  border-radius: 5px;
  border: none;

  &:hover {
    cursor: pointer;
    background-color: var(--color-white-100);
  }
}

.small {
  font-size: 0.8125rem;
}

.medium {
  padding: 5px 15px;
  font-size: 0.875rem;
}

.large {
  padding: 7px 21px;
  font-size: 0.9375rem;
}

.text {
  background-color: var(--color-white);
}

.contained {
  background-color: var(--color-main-blue);
  color: var(--color-white);

  &:hover {
    background-color: var(--color-main-blue-100);
  }
}

.outlined {
  border: solid var(--color-main-blue) 1px;

  &:hover {
    border-color: var(--color-main-blue-100);
  }
}
