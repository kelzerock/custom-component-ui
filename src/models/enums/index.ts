export { Checkbox } from "./enums-for-test/Checkbox";
export { Switch } from "./enums-for-test/Switch";
export { Modal } from "./enums-for-test/Modal";
