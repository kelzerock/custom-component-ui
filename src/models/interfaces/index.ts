export { CustomSelectProps } from "./custom-select-props";
export { CustomButtonProps } from "./custom-button-props";
export { CustomTextFieldProps } from "./custom-text-field-props";
export { OptionsData } from "./option-data";
