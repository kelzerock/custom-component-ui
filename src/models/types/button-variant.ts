export type ButtonVariant = "text" | "contained" | "outlined";
