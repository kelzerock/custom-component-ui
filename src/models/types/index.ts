export { ButtonSize } from "./button-size";
export { ButtonVariant } from "./button-variant";
