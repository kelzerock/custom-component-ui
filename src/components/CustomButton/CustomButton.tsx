import { FC } from "react";
import { CustomButtonProps } from "../../models/interfaces";
import styles from "./CustomButton.module.scss";
import clsx from "clsx";

export const CustomButton: FC<CustomButtonProps> = ({
  children,
  variant,
  size,
  className,
  ...props
}: CustomButtonProps) => {
  return (
    <button
      className={clsx(
        styles.button,
        variant && styles[variant],
        size && styles[size],
        className,
      )}
      {...props}
    >
      {children}
    </button>
  );
};
