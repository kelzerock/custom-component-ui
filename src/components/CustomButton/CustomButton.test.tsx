import { ButtonSize, ButtonVariant } from "../../models/types";
import { CustomButton } from "./CustomButton";
import { fireEvent, render, screen } from "@testing-library/react";

describe("test CustomButton component", () => {
  test("check custom props variant and size", () => {
    render(
      <CustomButton variant="contained" size="small">
        Click Me
      </CustomButton>,
    );

    const button = screen.getByRole("button");

    expect(button.textContent).toEqual("Click Me");
    expect(button.classList.contains("contained")).toBe(true);
    expect(button.classList.contains("small")).toBe(true);
    expect(button.classList.contains("button")).toBe(true);
  });

  test("check casual props for button, without text", () => {
    const variantStyle: ButtonVariant = "outlined";
    const sizeStyle: ButtonSize = "large";
    render(
      <CustomButton
        variant={variantStyle}
        size={sizeStyle}
        type="button"
        disabled
      />,
    );

    const button = screen.getByRole("button");

    expect(button.textContent).toEqual("");
    expect(button.classList.contains(variantStyle)).toBe(true);
    expect(button.classList.contains(sizeStyle)).toBe(true);
    expect(button.classList.contains("button")).toBe(true);
    expect(button.hasAttribute("disabled")).toBe(true);
    expect(button.getAttribute("disabled")).toEqual("");
    expect(button.getAttribute("type")).toEqual("button");
  });

  test("check onClick event", () => {
    const handleClick = jest.fn();
    render(<CustomButton onClick={handleClick}>Click</CustomButton>);

    fireEvent.click(screen.getByRole("button"));

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("check UI snapshot", () => {
    const { container } = render(<CustomButton>Snapshot</CustomButton>);
    expect(container).toMatchSnapshot();
  });
});
