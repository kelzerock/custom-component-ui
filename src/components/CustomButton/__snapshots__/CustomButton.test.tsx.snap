// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CustomButton component check UI snapshot 1`] = `
<div>
  <button
    class="button"
  >
    Snapshot
  </button>
</div>
`;
